java/ql/lib/semmle/code/java/security/SensitiveKeyboardCacheQuery.qll

```diff
@@ -0,0 +1,53 @@
+/** Definitions for the keyboard cache query */
+
+import java
+import semmle.code.java.security.SensitiveActions
+
+/** An Android Layout XML file. */
+class AndroidLayoutXmlFile extends XmlFile {
+  AndroidLayoutXmlFile() { this.getAbsolutePath().matches("%/res/layout/%.xml") }
```

Comment:
I think it'd be better to use `getRelativePath()` here instead. The chances are low, but let's just not risk `/res/layout/` appearing outside the project directory in the filesystem and skewing the results.